### PR TITLE
Clarify config settings for AS::halt_callback_chains_on_return_false

### PR DESCRIPTION
**Pull request by jonatack**
Despite the config name, it isn't super clear when to use true or false for this config and which of the two is the new default. I remember searching in the docs to make sure I got it right. Clarifying it here might help others to be sure they got it right, too.

[skip ci]

### CODE REVIEW

**Comment by sgrif** [comment]
Can we just add docs for the setting instead? I don't think we need to exhaustively document config options in the config file itself.

**Comment by jonatack** [comment]
OK, will do and update here.

**Comment by jonatack** [comment]
Moved docs from the initializer to the Configuring Guide.

Note that the guide previously indicated that
`ActiveSupport.halt_callback_chains_on_return_false` defaulted to `true`. This changes it to state that the default setting is `false`.

**Comment by jonatack** [comment]
Hm, the generated config file defaults to `false`, but if no setting is given it defaults to `true` as per the change log reproduced below. Will update the docs to clarify that.

    Newly generated Rails apps have a new initializer called
    `callback_terminator.rb` which sets the value of the configuration option
    `ActiveSupport.halt_callback_chains_on_return_false` to `false`.

    As a result, new Rails apps do not halt Active Record and Active Model
    callback chains when a callback returns `false`; only when they are
    explicitly halted with `throw(:abort)`.

    The terminator is *not* added when running `rake rails:update`, so returning
    `false` will still work on old apps ported to Rails 5, displaying a
    deprecation warning to prompt users to update their code to the new syntax.

**Comment by claudiob** [comment]
Your latest comment looks good :ribbon:

File: railties/lib/rails/generators/rails/app/templates/config/initializers/callback_terminator.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Be sure to restart your server when you modify this file.
 
-# Do not halt callback chains when a callback returns false. This is a new Rails 5.0 default, 
-# so introduced as a config to ensure apps made with earlier versions of Rails aren't affected when upgrading.
+# Do not halt callback chains when a callback returns false. This is a new
+# Rails 5.0 default, so it is introduced as a config to ensure that apps made
+# on earlier versions of Rails are not affected when upgrading.
```
Comment (matthewd) [inline railties/lib/rails/generators/rails/app/templates/config/initializers/callback_terminator.rb]: This seems like quite a story, particularly given that the only people who see this (new 5.0 apps) really shouldn't care about it at all.

Have we previously been this verbose on similar config values?

(This is more of a general observation about the comment as it already stood, unrelated to your specific changes.)

**Comment by jonatack** [comment]
@claudiob I hesitate because technically, the default is currently just `true` during the deprecation period, and will switch to `false`. Your change log rocks and is the clearest version to me.

File: railties/lib/rails/generators/rails/app/templates/config/initializers/callback_terminator.rb (hunk 1 above)
Comment (jonatack) [inline railties/lib/rails/generators/rails/app/templates/config/initializers/callback_terminator.rb]: There is the same addition in `active_record_belongs_to_required_by_default.rb`.

This change could be removed from this PR and added to another one for both config files.

Comment (jonatack) [inline railties/lib/rails/generators/rails/app/templates/config/initializers/callback_terminator.rb]: The changes were in 569e03d

**Comment by claudiob** [comment]
Ok then fine with closing it :grin:

**Comment by rafaelfranca** [comment]
Closing them.

**Comment by jonatack** [comment]
I just updated the PR. The configuration still needs to be explained in the docs.

**Pull request merged**